### ISSUE TITLE
Simplify generating passwords with random category distribution

### ISSUE THREAD

**Issue by TjeuKayim** [Keeper-Security/Commander#846]
When using `keepercommander.generator` as a library to generate passwords, the current design makes it easy to accidentally use lower entropy than intended.

Let's say I want passwords with 20 characters, including at least 1 character from each category: uppercase, lowercase, digits, symbols. Here are three ways to do that:

```py
from keepercommander.generator import generate, KeeperPasswordGenerator
import string

password1 = generate(20)
# entropy = log2(26**5 * 26**5 * 10**5 * 32**5) = 88.6 bits

password2 = KeeperPasswordGenerator(20).generate()
# entropy = log2(26**5 * 26**5 * 10**5 * 20**5) = 85.2 bits
# (only uses the 20 symbols from PW_SPECIAL_CHARACTERS)

password3 = KeeperPasswordGenerator(20, 1, 1, 1, 1, string.punctuation).generate()
# entropy = log2(26 * 26 * 10 * 32 * (32+26+26+10)**16) = 122.6 bits
```

The first 2 methods generate characters in the following distribution:

- 5 lowercase letters
- 5 uppercase letters
- 5 digits
- 5 punctuation characters

Intuitively, you might expect that a random alphanumeric+punctuation password of length 20 with 5 from each category, is more secure than a random alphanumeric (without punctuation) password of length 15 (with random category distribution). Though the opposite is true.

The third method uses the following distribution (resulting in better entropy):

- 1-17 lowercase letters
- 1-17 uppercase letters
- 1-17 digits
- 1-17 punctuation characters

Keeping the current behavior is useful when a service requires at least 2 punctuation characters in the password, or alike.
On the other hand, one now has to thoroughly read the code to understand what parameters to pass for a maximum entropy. I would expect this library to have a more straightforward way to create a more random category distribution, or at least contain documentation comments explaining how the fixed category distribution behavior affects the entropy.

**Comment by craiglurey** [comment]
Thank you for opening this issue.  We will investigate this and revert back to you with a solution.

**Comment by sk-keeper** [comment]
This issue has been fixed in Commander release v16.7.6
